server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -581,6 +598,13 @@ protected DateFieldMapper clone() {
 
     @Override
     protected void parseCreateField(ParseContext context) throws IOException {
+        if (singletonDataStreamTimestamp) {
+            if (context.isDataStreamTimestampParsed()) {
+                throw new IllegalArgumentException("timestamp field has multiple values, only a single value is allowed");
```

Comment:
Small comment, this could mention 'data stream timestamp' for clarity. We also try to include the field name when possible to help with debugging: "Encountered data stream timestamp field [my-timestamp] with multiple values ..."